mathext/hyp2f1_test.go

```diff
@@ -0,0 +1,230 @@
+			v := eqn15_2_10(0.1, 0.44, 0.611, test.x)
+			if math.Abs(v) > 1e-6 {
+				t.Errorf("%f %f", test.x, v)
+			}
+		})
+	}
+}
+
+func TestHyp2f1_15_1(t *testing.T) {
+	t.Parallel()
+
+	type equation struct {
+		name string
+		f    func(float64) float64
+	}
+
+	equations := []equation{
+		{
+			name: "15_1_3",
+			f: func(z float64) float64 {
+				lhs := h(1, 1, 2, z)
+				rhs := -math.Log(1-z) / z
+				return math.Abs(rhs - lhs)
+			},
+		},
+		{
+			name: "15_1_5",
+			f: func(z float64) float64 {
+				lhs := h(1./2, 1, 3./2, -z*z)
+				rhs := math.Atan(z) / z
+				return math.Abs(rhs - lhs)
+			},
+		},
+		{
+			name: "15_1_7a",
+			f: func(z float64) float64 {
+				lhs := h(1./2, 1./2, 3./2, -z*z)
+				rhs := math.Sqrt(1+z*z) * h(1, 1, 3./2, -z*z)
+				return math.Abs(rhs - lhs)
+			},
+		},
+		{
+			name: "15_1_7b",
+			f: func(z float64) float64 {
+				lhs := h(1./2, 1./2, 3./2, -z*z)
+				rhs := math.Log(z+math.Sqrt(1+z*z)) / z
+				return math.Abs(rhs - lhs)
+			},
+		},
+	}
+
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			t.Parallel()
+			for _, eqn := range equations {
+				if d := eqn.f(test.x); d > 1e-10 {
+					t.Errorf("%s %f %f", eqn.name, test.x, d)
+				}
+			}
+		})
+	}
+}
+
+func TestHyp2f1_15_1_zz(t *testing.T) {
+	t.Parallel()
+
+	type equation struct {
+		name string
+		f    func(float64) float64
+	}
+
+	equations := []equation{
+		{
+			name: "15_1_4",
+			f: func(z float64) float64 {
+				lhs := h(1./2, 1, 3./2, z*z)
+				rhs := 0.5 * math.Log((1+z)/(1-z)) / z
+				return math.Abs(rhs - lhs)
+			},
+		},
+		{
+			name: "15_1_6a",
+			f: func(z float64) float64 {
+				lhs := h(1./2, 1./2, 3./2, z*z)
+				rhs := math.Sqrt(1-z*z) * h(1, 1, 3./2, z*z)
+				return math.Abs(rhs - lhs)
+			},
+		},
+		{
+			name: "15_1_6b",
+			f: func(z float64) float64 {
+				lhs := h(1./2, 1./2, 3./2, z*z)
+				rhs := math.Asin(z) / z
+				return math.Abs(rhs - lhs)
+			},
+		},
+	}
+
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			t.Parallel()
+
+			if test.x*test.x > 1 {
+				return
+			}
+
+			for _, eqn := range equations {
+				if d := eqn.f(test.x); d > 1e-10 {
+					t.Errorf("%s %f %f", eqn.name, test.x, d)
+				}
+			}
+		})
+	}
+}
+
+func TestHyp2f1_Igor_Kojanov(t *testing.T) {
+	t.Parallel()
+
+	var y float64
+	y = h(1, 2, 3, 0)
+	if d := y - 1; d != 0 {
+		t.Errorf("%f %f", y, d)
+	}
+
+	y = h(1, 1.64, 2.64, -0.1111)
+	if d := y - 0.9361003540660249866434; math.Abs(d) > 1e-15 {
+		t.Errorf("%f %f", y, d)
+	}
+}
+
+func TestHyp2f1_John_Ormerod(t *testing.T) {
+	t.Parallel()
+
+	y := h(5.25, 1, 6.5, 0.501)
+	if d := y - 1.70239432012007391092082702795; math.Abs(d) > 1e-10 {
+		t.Errorf("%f %f", y, d)
+	}
+}
+
+func h(a, b, c, x float64) float64 {
+	y, err := Hyp2f1(a, b, c, x)
+	if err != nil {
+		panic(err)
+	}
+	return y
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	log.SetFlags(log.Lmicroseconds | log.Llongfile | log.LstdFlags)
+
+	m.Run()
+}
```

Comment:
Please don't do this.

Reply:
Sorry, should have removed this before committing.